common/src/main/java/uk/gov/justice/services/common/converter/TypedConverter.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.services.common.converter;
+
+/**
+ * A converter converts a source object of type S to a target of type T.
+ */
+@FunctionalInterface
+public interface TypedConverter<S, T> {
+
+    /**
+     * Convert the source object of type S to target type T.
+     *
+     * @param source the source object to convert, which must be an instance of S (never null)
+     * @return the converted object, which must be an instance of T (potentially null)
+     * @throws IllegalArgumentException - if the source cannot be converted to the desired target type
+     */
+    <R extends T> R convert(S source, Class<R> clazz);
```

Comment:
I was told not to do it in interface, but we are doing at other places so i will make them final for consistency.